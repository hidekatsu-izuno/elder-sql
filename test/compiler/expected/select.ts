/** 
 * 
 * List employees
 */
export function listEmployees(engine, params) {
  let text = ""
  const args = new Array<any>()
  const ctx0 = {...params}
  text += "\nselect * from employee where"
  if (sandbox(ctx0, "employeeId != null")) {
    text += "\n  employee_id = ?"
    args.push(sandbox(ctx0, "employeeId"))
    if (sandbox(ctx0, "employeeName != null")) {
      text += "\n    and\n    employee_name = ?"
      args.push(sandbox(ctx0, "employeeName"))
    } else {
      text += "\n    and\n    employee_name is null"
    }
  }
  return engine.execute(text, args)
}
function sandbox(ctx, expr) {
  return (new Function("context", "with (ctx) { return (" + expr + ") }"))(ctx)
}