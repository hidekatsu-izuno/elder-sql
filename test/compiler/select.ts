export const actual = `
/*#define listEmployees
 * 
 * List employees
 */
select * from employee where
/*#if employeeId != null */
  employee_id = /*#{employeeId}*/99
  /*#if employeeName != null */
    and employee_name in /*#{employeeName}*/('hoge')
  /*#elif employeeName === "test"*/
--* and
--* employee_name = /*\${subquery}*/
  /*#else*/
--* and
--* employee_name is null
  /*#end*/

  /*#if employeeTypes.length > 0*/
    and (
    /*#for employeeType : employeeTypes */
      or employee_type = /*#{employeeType}*/
    /*#end*/
    )
  /*#end*/
/*#end*/
`.trim()
export const expected = `
/** 
 * 
 * List employees
 */
export function listEmployees(engine, params) {
  const text = ""
  const args = []
  const context0 = {...params}
  text += "\nselect * from employee where"
  if (employeeId != null) {
    text += "\n  employee_id = ?"
    args.push(sandbox(context0, "employeeId"))
    if (employeeName != null) {
      text += "\n    and\n    employee_name = ?"
      args.push(sandbox(context0, "employeeName"))
    } else {
      text += "\n    and\n    employee_name is null"
    )
  )
  return engine.execute(text, args)
}
function sandbox(context, expr) {
  return (new Function("context", "with (context) " + expr))(context)
}
`.trim()