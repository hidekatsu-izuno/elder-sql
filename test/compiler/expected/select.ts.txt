/**
 * 
 * List employees
 */
export function listEmployees(engine, params) {
  let text = "";
  const args = [];
  const ctx0 = { ...params };
  text += "select* from employee where";
  if (sandbox(employeeId != null)) {
    text += "\n  employee_id= :employeeId  and employee_id= :employeeId  ";
    if (sandbox(employeeName != null)) {
      text += "\n    and employee_name in :employeeId    or employee_name LIKE :employeeName  /*#elsif employeeName === \"test\"*/\n    and    employee_name= :employeeId    or    employee_name={#= \"'\"+ employeeName+ \"'\"}  ";
    } else {
      text += "\n    and    employee_name is null  ";
    }
    text += "\n\n  and(  ";
    (employeeTypes).forEach((employeeType) => {
      const ctx1 = {...ctx0, employeeType};
      text += "\n    or employee_type= :employeeType  ";
    })
    text += "\n  )";
  }
  return engine.execute(text, args);
}

